xDrip/Scenes/Entries/EditFoodEntry/EditFoodEntryInteractor.swift

```diff
@@ -11,22 +11,83 @@
 //
 
 import UIKit
+import RealmSwift
 
 protocol EditFoodEntryBusinessLogic {
     func doLoad(request: EditFoodEntry.Load.Request)
+    func doCancel(request: EditFoodEntry.Cancel.Request)
+    func doSave(request: EditFoodEntry.Save.Request)
 }
 
-protocol EditFoodEntryDataStore: AnyObject {    
+protocol EditFoodEntryDataStore: AnyObject {
+    var entryType: EditFoodEntry.EntryType? { get set }
 }
 
 final class EditFoodEntryInteractor: EditFoodEntryBusinessLogic, EditFoodEntryDataStore {
     var presenter: EditFoodEntryPresentationLogic?
     var router: EditFoodEntryRoutingLogic?
     
+    var entryType: EditFoodEntry.EntryType?
+    private lazy var carbEntry: CarbEntry = {
+        return CarbEntry()
+    }()
+    private lazy var bolusEntry: BolusEntry = {
+        return BolusEntry()
+    }()
+    
     // MARK: Do something
     
     func doLoad(request: EditFoodEntry.Load.Request) {
-        let response = EditFoodEntry.Load.Response()
+        let response = EditFoodEntry.Load.Response(
+            entryType: entryType ?? .food,
+            textChangedHandler: handleTextChanged(_:_:),
+            dateChangedHandler: handleDateChanged(_:_:),
+            foodTypeChangedHandler: handleFoodTypeChanged(_:)
+        )
         presenter?.presentLoad(response: response)
     }
+    
+    func doCancel(request: EditFoodEntry.Cancel.Request) {
+        router?.dismissScene()
+    }
+    
+    func doSave(request: EditFoodEntry.Save.Request) {
+        if bolusEntry.amount != 0.0 {
```

Comment:
The logic seems wrong here. There shouldn't be a relation between `.amount != 0` and adding object to Realm.